File: NCoreUtils.Images.GoogleCloudStorage/GoogleObjectData.fs
namespace NCoreUtils.Images.GoogleCloudStorage

open System.Text.Json.Serialization

[<CLIMutable>]
[<NoEquality; NoComparison>]
type GoogleObjectData = {
  [<property: JsonPropertyName("cacheControl")>]
  CacheControl : string
  [<property: JsonPropertyName("contentType")>]
  ContentType  : string
  [<property: JsonPropertyName("name")>]
  Name         : string }

File: NCoreUtils.Images.GoogleCloudStorage/GoogleCloudStorage/GoogleObjectData.cs
using System.Text.Json.Serialization;

namespace NCoreUtils.Images.GoogleCloudStorage
{
    public class GoogleObjectData
    {
        [JsonPropertyName("cacheControl")]
        public string? CacheControl { get; set; }

        [JsonPropertyName("contentType")]
        public string? ContentType { get; set; }

        [JsonPropertyName("name")]
        public string? Name { get; set; }
    }
}